Stock.Api/Controllers/ProviderController.cs

```diff
@@ -35,14 +36,19 @@ public ProviderController(ProviderService service, IMapper mapper)
         [HttpGet]
         public ActionResult<IEnumerable<ProviderDTO>> Get()
         {
+            logger.LogInformation("Mensaje Log de Get");
             try
             {
-                var result = service.GetAll();
-                return mapper.Map<IEnumerable<ProviderDTO>>(result).ToList();
+                var provider = service.GetAll();
+                if (provider == null)
+                {
+                    return Ok(new { statusCode = "404", result = "No hay datos en Proveedores" });
```

Comment:
Va mejor lo de devolver el not found, pero la idea es que sea el HttpStatus correcto, si haces un `return Ok();` el response va a ser 200 OK on el json que dice "404".
En lugar de eso, con hacer simplemente `return NotFound();` ya estas haciendo que el response sea 404 Not Found y no es necesario aclarar nada mas en el body del response

Reply:
De acuerdo, ya lo cambié